src/transforms/at-range.js

```diff
@@ -113,14 +113,23 @@ Transforms.deleteAtRange = (transform, range, options = {}) => {
   const endBlock = document.getClosestBlock(document.getNextText(endKey).key)
 
   if (startBlock.key !== endBlock.key) {
-    endBlock.nodes.forEach((child, i) => {
-      const newKey = startBlock.key
-      const newIndex = startBlock.nodes.size + i
+    let blockToRemove = endBlock
+    let blockToKeep = startBlock
+
+    // We keep the style of the blockToKeep. endBlock should keep style if deleting empty line.
+    if (startBlock.isEmpty && !endBlock.isEmpty) {
```

Comment:
I actually think this should probably be just `if (startBlock.isEmpty)`.

When you have:

```
<blank p>
<blank h1>
```

And you delete on the `h1` should the new node be `h1` or `p`? I'm thinking `h1`...